doc/sphinx-guides/source/developers/containers.rst

```diff
@@ -194,6 +194,115 @@ Once you have multiple Glassfish servers you may notice bugs that will require a
 
 Multiple PostgreSQL servers are possible within the OpenShift environment as well and have been set up with some amount of replication. "dataverse-postgresql-0" is the master and non-zero pods are the slaves. We have just scratched the surface of this configuration but replication from master to slave seems to we working. Future work could include failover and making Dataverse smarter about utilizing multiple PostgreSQL servers for reads. Right now we assume Dataverse is only being used with a single PostgreSQL server and that it's the master.
 
+Solr supports index distribution and replication for scaling. For OpenShift use, we choose replication. It's possible to scale up solar using the method method similar to Glassfish, as mentioned aboved
+In OpenShift, the first Solr pod, dataverse-solr-0, will be the master node, and the rest will be slave nodes
+
+
+Configuring Persistent Volumes and Solr master node recovery 
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Solr requires backing up the search index to a storage. For our proof of concept, we configure a hostPath, which allows solr containers to access the hosts' file system, for our Solr containers backups. To read more about OpenShift/Kubernetes' persistent volumes, please visit https://kubernetes.io/docs/concepts/storage/persistent-volumes
+To allow containers to use host's storage, we need to allow access to that directory first. In this example, we expose /tmp/share to the containers:
+```
+mkdir /tmp/share           # Create a directory that could be mounted in container. This is mounted as /share in container.
+chcon -R -t svirt_sandbox_file_t /tmp/share/ # Change selinux label so that containers can read/write from/to directory.
+chmod 777 -R /tmp/share
+oc login -u system:admin
+oc edit scc restricted     # Update allowHostDirVolumePlugin to true and runAsUser type to RunAsAny
+```
+
+To add a persistent volume and persistent volume claim, in conf/docker/openshift/openshift.json, add the following to objects in openshift.json:
```

Comment:
Doc this as for development purposes since it's using hostPath.  And that you would need to use a different kind of PV for a production use case.